haskell/src/Lentille/GitHub/RateLimit.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# OPTIONS_GHC -Wno-overlapping-patterns #-}
+
+module Lentille.GitHub.RateLimit where
+
+import Data.Morpheus.Client
+import Lentille (MonadGraphQL)
+import Lentille.GraphQL
+import Monocle.Prelude
+
+newtype DateTime = DateTime Text deriving (Show, Eq, EncodeScalar, DecodeScalar)
+
+defineByDocumentFile
+  ghSchemaLocation
+  [gql|
+    query GetRateLimit  {
+      rateLimit {
+        used
+        remaining
+        resetAt
+      }
+    }
+  |]
+
+transformResponse :: GetRateLimit -> RateLimit
+transformResponse = \case
+  GetRateLimit
+    ( Just
+        (RateLimitRateLimit used remaining (DateTime resetAt))
+      ) -> RateLimit {..}
+  respOther -> error ("Invalid response: " <> show respOther)
+
+getRateLimit ::
+  (FromJSON a, Fetch a, MonadGraphQL m) =>
+  GraphClient ->
+  m (Either (FetchError a) a, [ReqLog])
+getRateLimit client = fetchWithLog (doGraphRequest client) ()
```

Comment:
the '()' here is the issue. However it seems that's how it is handled in the example: https://github.com/morpheusgraphql/morpheus-graphql/blob/master/examples/client/src/Client/StarWarsClient.hs#L60

Reply:
@TristanCacqueray any idea ?

Reply:
The unit `()` is indeed a Fetch Instance for RateLimit:

```
> :info Args
type Fetch :: * -> Constraint
type instance Args GetRateLimit = ()
```

But GHC does not know you want a GetRateLimit, and the error indicates that unit is not an instance of `Fetch a`. Thus the solution is to help the type checker by using this annotation:

`MonadGraphQL m => GraphClient -> m (Either (FetchError GetRateLimit) GetRateLimit, [ReqLog])`

Reply:
Alright that make sense that from my function signature nothing provided info about the GetRateLimit data type. Thanks for the quick insight !